platform-api/src/dependencies.py

```
@@ -207,21 +207,77 @@ async def get_sanity(settings: WorkerSettings = Depends(get_settings)) -> Sanity
+
+        role_clause = (
+            "AND user.role = ?" if require_role else "AND user.role IN ('sponsor')"
+        )
+        query = f"""
+            SELECT user.email, user.role
+            FROM session
+            JOIN user ON session.user_id = user.id
+            WHERE session.token = ?
+              {role_clause}
+              AND session.expires_at > (unixepoch() * 1000)
+        """
```

Ruff (0.15.12)

[error] Possible SQL injection vector through string-based query construction

(S608)